From dcd37d9548049a9764a4286dd238f6731826e8ee Mon Sep 17 00:00:00 2001
From: Richard Hansen <rhansen@rhansen.org>
Date: Sun, 16 Apr 2023 23:19:27 -0400
Subject: [PATCH] Initialize WaylandSessionPrivate.vt to -1

This matters when starting a Wayland session on a non-seat0 seat:
  * It eliminates an erroneous attempt to switch to VT 0.
  * The XDG_VTNR environment variable is no longer set.

Also add a multiseat test.

A similar change for XServerLocalPrivate is not needed because its vt field is
already initialized to -1.
---
 src/wayland-session.c                 |  2 +
 4 files changed, 70 insertions(+)

diff --git a/src/wayland-session.c b/src/wayland-session.c
index 187bf1f07..b2dc8201c 100644
--- a/src/wayland-session.c
+++ b/src/wayland-session.c
@@ -78,6 +78,8 @@ wayland_session_disconnect_session (DisplayServer *display_server, Session *sess
 static void
 wayland_session_init (WaylandSession *session)
 {
+    WaylandSessionPrivate *priv = wayland_session_get_instance_private (session);
+    priv->vt = -1;
 }
 
 static void
